src/_blog/cryptotls.md

```diff
@@ -0,0 +1,60 @@
+2. Every time there was a secruity-related fix in crypto/tls, we had to ask our users to update to a new (patch) release of quic-go as well.
+3. Since we couldn’t know how crypto/tls would look in future Go versions, we had to restrict the Go versions that could be used to build quic-go. This lack of forwards-compatibility prompted regular complaints from users who weren’t familiar with the internals of quic-go.
+
+This was not an ideal situation, but given the nature of QUIC, we had little choice.
+
+## Solving the Problem once and for all
+
+To tackle this situation, we joined forces with Filippo Valsorda, a former member of Google’s Go team. Although he has left the company since, he’s still maintaining crypto/tls and some other crypto packages in the standard library. Filippo, now a full-time maintainer, is sponsored by Protocol Labs for his remarkable [open-source work](https://words.filippo.io/full-time-maintainer/).
+
+Our first joint endeavor was to establish an API for crypto/tls that would enable QUIC implementations to use crypto/tls, detailed in this [GitHub issue](https://github.com/golang/go/issues/44886). This new API was designed to only support the normal (1-RTT) QUIC handshake, not the 0-RTT handshake (more on that below). 0-RTT allows clients to resume connections to servers they have previously connected to, and to send application data right in the first flight.
+
+After long discussions (both over Zoom and on the GitHub issue), we arrived at a proposal that was much cleaner than our homegrown qtls API.
+
+Adding support for 0-RTT was a fairly large endeavor. This is because the TLS and the QUIC stack need to coordinate quite a lot to enable this feature. For example, both the client and the server need to remember certain configuration parameters (called QUIC transport parameters) from the original QUIC connection. Among others, these transport parameters include values like flow control windows (i.e. how many bytes a client is allowed to send on a newly established stream) and how many streams the client is allowed to open. This information is needed for the client to stay within these limits when resuming a connection. The server typically encrypts these values and stores them in the session ticket. When the client restores the session, it sends the session ticket to the server (as part of the TLS ClientHello message), allowing the server to restore the transport parameters (without having to store them at all!). Of course, the client also needs to store some its parameters alongside the ticket, so it can restore them when resuming the session.
+
+This means that crypto/tls needed to support adding data to session tickets, both the client and server sides. It turned out that the API required for this would solve also solve a large number of other longstanding [issues related to session tickets](https://github.com/golang/go/issues/60105).
```

Comment:
Make this the same paragraph as the previous one. It's a continuation of the previous sentence.